### PR TITLE
[fix](nestedType)fix nested data type to create table

### PR DESCRIPTION
**Pull request by zfr9527**
## Proposed changes

fix nested data type to create table

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

File: regression-test/plugins/plugins_create_table_nested_type.groovy [hunk 1]
```diff
@@ -18,75 +18,125 @@
 import org.apache.doris.regression.suite.Suite
 
 // create table with nested data type, now default complex data include array, map, struct
-Suite.metaClass.create_table_with_nested_type = { int maxDepth, String tbName /* param */ ->
+Suite.metaClass.create_table_with_nested_type = { int maxDepth, def typeArr, String tbName /* param */ ->
     Suite suite = delegate as Suite
 
+    if (typeArr.size() != maxDepth) {
+        println("level not equal typeArr")
+        return
```
Comment (shuke987) [inline regression-test/plugins/plugins_create_table_nested_type.groovy:26]: throw exception？

File: regression-test/plugins/plugins_create_table_nested_type.groovy [hunk 2]
```diff
@@ -18,75 +18,125 @@
...
+//                           "char(15)", "varchar(100)", "text", "hll","bitmap", "QUANTILE_STATE"]
     def dataTypeArr = ["boolean", "tinyint(4)", "smallint(6)", "int(11)", "bigint(20)", "largeint(40)", "float",
                        "double", "decimal(20, 3)", "decimalv3(20, 3)", "date", "datetime", "datev2", "datetimev2(0)",
                        "char(15)", "varchar(100)", "text"]
     def colNameArr = ["c_bool", "c_tinyint", "c_smallint", "c_int", "c_bigint", "c_largeint", "c_float",
                       "c_double", "c_decimal", "c_decimalv3", "c_date", "c_datetime", "c_datev2", "c_datetimev2",
                       "c_char", "c_varchar", "c_string"]
     def complexDataTypeArr = ["array", "map", "struct"]
-//    def tbName = "test"
+    def baseStructScala = "col1:int(11),col2:tinyint(4),col3:smallint(6),col4:boolean,col5:bigint(20),col6:largeint(40)," +
```
Comment (shuke987) [inline regression-test/plugins/plugins_create_table_nested_type.groovy:42]: use cross-line string

File: regression-test/plugins/plugins_create_table_nested_type.groovy (hunk 1 above)
Comment (zfr9527) [reply regression-test/plugins/plugins_create_table_nested_type.groovy:30]: Manual calls to the old code would have made errors, and the current code is guaranteed not to make errors here

File: regression-test/plugins/plugins_create_table_nested_type.groovy (hunk 2 above)
Comment (zfr9527) [reply regression-test/plugins/plugins_create_table_nested_type.groovy]: Have been modified

**Comment by xiaokang** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by xiaokang** [comment]
run buildall

**Comment by xiaokang** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**